### ISSUE TITLE
MM_anaerobic_CO2_H2 and MM medium enriched with additional compunds lead to differentent growth capacity  

### ISSUE THREAD

**Issue by nicola-debernardini** [jotech/gapseq#89]
Hi, I'm performing the gap filling of _Methanothermobacter wolfeii_ - [LT608329.1](https://www.ncbi.nlm.nih.gov/nuccore/LT608329) using the minimal medium "MM_anaerobic_CO2_H2" as shown in the issue [54](https://github.com/jotech/gapseq/issues/54) but I obtain different results when try to add compounds to that medium. 

1. gapfilling on MM:
When I perform the gapfilling on MM Methanothermobacter  grows (0.089)

![image](https://user-images.githubusercontent.com/45240213/126180450-9186e2a1-1de5-4dec-9241-0b4ea467f193.png)

The medium used is: [MM_anaerobic_CO2_H2.csv](https://github.com/jotech/gapseq/files/6842019/MM_anaerobic_CO2_H2.csv)

2. gapfilling on MM plus other compounds:
When I add all the compounds of the ALLmed to the MM Methanothermobacter does not grow anymore.
 
![image](https://user-images.githubusercontent.com/45240213/126180356-683417b9-2032-444d-a9b5-a7b704f1b842.png)

The medium used has this composition: [MM_anaerobic_CO2_H2_plus_ALLmed.csv](https://github.com/jotech/gapseq/files/6842024/MM_anaerobic_CO2_H2_plus_ALLmed.csv)

Do you have any clue why does this happen? 
Why does the same reference genome gapfilled on medium containing the same compounds and with the same uptake bounds lead to difference growth rates when I add compounds to one of them (in this case not growing at all)?

The models were gapfilled with version (v.1.1 6ae3a6a) - [models.zip](https://github.com/jotech/gapseq/files/6842198/models.zip).

I appreciate any help you can provide.

Best regards,
Nicola

**Comment by Waschina** [comment]
Hi Nicola,
this looks odd, indeed. I ran the whole workflow but can't reproduce the issue.
The exact workflow I used:

```bash
model="LT608329.1"

gapseq find -p all -m Archaea -t Archaea $model.fna.gz
gapseq find-transport $model.fna.gz

gapseq draft -r $model-all-Reactions.tbl \
-t $model-Transporter.tbl \
-p $model-all-Pathways.tbl \
-c $model.fna.gz \
-u 200 -l 100

# gapfill with minimal media
gapseq fill -m $model-draft.RDS \
-n MM_anaerobic_CO2_H2.csv \
-c $model-rxnWeights.RDS \
-g $model-rxnXgenes.RDS \
-b 100 -e highH2

# gapfill with minimal media plus ALLmed
gapseq fill -m $model-draft.RDS \
-n MM_anaerobic_CO2_H2_plus_ALLmed.csv \
-c $model-rxnWeights.RDS \
-g $model-rxnXgenes.RDS \
-b 100 -e highH2
```

With the **minimal medium** I get:
![image](https://user-images.githubusercontent.com/32539757/126486692-425dcf58-079b-402c-bd57-b68b52f06068.png)

With the **extended medium**:
![image](https://user-images.githubusercontent.com/32539757/126486785-9b7c9a38-ce02-4981-8e1e-2814331f2c38.png)


I am wondering if this could be due to different LP-solvers. In my case I used cplex version 1210. What LP-solver are you using?

Best
Silvio

**Comment by nicola-debernardini** [comment]
Hi Silvio,

Many thanks for your answer. I realized I used the version 1.1 **6ae3a6a** to run only the last step ("fill") and version 1.1 **d6522aa** to run "find", "find-transport" and "draft". You probably solved the problem in the latter versions of one of these three commands.  

I tried to run your procedure using version 1.1 **d6522aa** in all the steps and the results I obtained are **identical** to yours. The solver I used is cplex version 12.8 but the different version of the solver didn't cause any change.

Also, thanks for sharing you procedure. I never used the "**-m**" option during the "find" step but I saw it runs much faster when I used it. 

Thanks again for the answer!

Cheers,
Nicola
